crud.go

```diff
@@ -80,6 +80,58 @@ func (cc *CrudComponent) Get(ctx context.Context, opts *GetOptions) (*GetResult,
 		})
 }
 
+type GetReplicaOptions struct {
+	Key            []byte
+	ScopeName      string
+	CollectionName string
+	ReplicaIdx     uint32 // TODO(chvck): Check if this should be an uint32
+}
+
+type GetReplicaResult struct {
+	Value    []byte
+	Flags    uint32
+	Datatype uint8
+	Cas      uint64
+}
+
+func (cc *CrudComponent) GetReplica(ctx context.Context, opts *GetReplicaOptions) (*GetReplicaResult, error) {
+	fn := func(collectionID uint32, manifestID uint64, endpoint string, vbID uint16, client KvClient) (*GetReplicaResult, error) {
+		resp, err := client.GetReplica(ctx, &memdx.GetReplicaRequest{
+			CollectionID: collectionID,
+			Key:          opts.Key,
+			VbucketID:    vbID,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		return &GetReplicaResult{
+			Value:    resp.Value,
+			Flags:    resp.Flags,
+			Datatype: resp.Datatype,
+			Cas:      resp.Cas,
+		}, nil
+	}
+
+	return OrchestrateMemdRetries(
+		ctx, cc.retries,
+		func() (*GetReplicaResult, error) {
+			return OrchestrateMemdCollectionID(
+				ctx, cc.collections, opts.ScopeName, opts.CollectionName,
+				func(collectionID uint32, manifestID uint64) (*GetReplicaResult, error) {
+					// TODO(chvck): this needs to handle retries and whatnot.
```

Comment:
I just felt like doing this and passing 0 everywhere bar 1 or 2 cases was a bit gross.

Reply:
Fair enough.  I guess we can have a generic `OrchestrateMemdReplicaRouting` which takes the replica index, and then a `OrchestrateMemdRouting` which calls the former with replica index 0 to avoid duplication.